packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -109,76 +54,430 @@ const invalidateCustomerBasketsQuery = (
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addItemToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addPaymentInstrumentToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addPriceBooksToBasket(customerId, {parameters}) {
         // TODO: Convert invalidate to an update that removes the matching basket
-        ...invalidateCustomerBasketsQuery(customerId, parameters),
-        remove: [
-            and(
-                matchParameters(toGetBasketParameters(parameters)),
-                matchesPath(getBasketPath(parameters))
-            )
-        ]
-    }),
-    mergeBasket: updateBasketWithResponseBasketId,
-    removeCouponFromBasket: updateBasket,
-    removeGiftCertificateItemFromBasket: TODO('removeGiftCertificateItemFromBasket'),
-    removeItemFromBasket: updateBasket,
-    removePaymentInstrumentFromBasket: updateBasket,
-    removeShipmentFromBasket: TODO('removeShipmentFromBasket'),
-    transferBasket: TODO('transferBasket'),
-    updateBasket: updateBasket,
-    updateBillingAddressForBasket: updateBasket,
-    updateCustomerForBasket: updateBasket,
-    updateGiftCertificateItemInBasket: TODO('updateGiftCertificateItemInBasket'),
-    updateItemInBasket: updateBasket,
-    updatePaymentInstrumentInBasket: updateBasket,
-    updateShipmentForBasket: TODO('updateShipmentForBasket'),
-    updateShippingAddressForShipment: updateBasket,
-    updateShippingMethodForShipment: updateBasket
+        const invalidations = invalidateCustomerBasketsQuery(customerId, parameters)
+        invalidations?.invalidate?.push({queryKey: getPriceBooksForBasket.queryKey(parameters)})
```

Comment:
This will only invalidate `getPriceBooksForBasket` if `invalidations.invalidate` exists, i.e. only if `customerId` is defined. It doesn't seem the price books endpoint relies on `customerId`, so should there be a fallback for `invalidations`?